modules/aggregation-layers/src/contour-layer/marching-squares.js

```diff
@@ -162,12 +188,35 @@ export function getVertices(opts) {
 
   // default case is ISO_LINES
   const lines = [];
-  offsets.forEach(xyOffsets => {
-    xyOffsets.forEach(offset => {
-      const vX = refVertexX + offset[0] * cellSize[0];
-      const vY = refVertexY + offset[1] * cellSize[1];
+  offsets.forEach((xyOffsets, i) => {
+    xyOffsets.forEach((offset, j) => {
+      let xOffset;
+      let yOffset;
+      if (isLI) {
+        const li = getSmoothOffset(linearOffsets[i][j], weights, thresholdData.threshold);
+        xOffset = !offset[0] ? offset[0] + li : offset[0];
```

Comment:
Add a comment here, why are we doing `!offset[0]`?

Reply:
The offset x and y axis contains value 0, 0.5, - 0.5. As I have to add the interpolation to offset having value 0 only. So I added this checks `!offset[0]?` to check whether the value is 0 or other than 0.
example - consider the offset [0, 0.5]. In this case, the Interpolation should be added to offset[0] and not to offset[1].